File: 백준/Silver/2559. 수열/수열.py
import sys
input = sys.stdin.readline

n,k = map(int,input().split())
nums = list(map(int,input().split()))

prefix = [0 for _ in range(n+1)]

for i in range(1, n+1):
    prefix[i] = prefix[i-1] + nums[i-1]

answer = []
for i in range(k,n+1):
    answer.append(prefix[i] - prefix[i-k])

print(max(answer))

File: 백준/Silver/2559. 수열/수열.java
import java.io.*;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());

        int[] temperatures = new int[n];

        st = new StringTokenizer(br.readLine());
        for(int i=0; i<n; i++) {
            temperatures[i] = Integer.parseInt(st.nextToken());
        }

        int[] prefix = new int[n+1];
        for(int i=1; i<n+1; i++) {
            prefix[i] = prefix[i-1] + temperatures[i-1];
        }

        int maxSum = Integer.MIN_VALUE;

        for(int i=k; i<=n; i++) {
            int sum = prefix[i] - prefix[i-k];
            maxSum = Math.max(maxSum, sum);
        }

        System.out.println(maxSum);
    }
}